FilePersistence/src/version_control/Commit.cpp

```diff
@@ -108,22 +108,24 @@ bool Commit::isValidMatch(const char* content, qint64 size, const char* indexOfI
 {
 	start = indexOfId-content;
 	end = start;
+	int precedingClosingBraceCount = 0;
 	// start is the first character of the line containing id
 	while (start >= 0 && content[start] != '\n')
 	{
-		// String is of the form {.*} {id}
-		if (!findChildrenByParentId && content[start] == '}') return false;
+		// Remove fake-ids
+		if (content[start] == '}')
+		{
+			precedingClosingBraceCount++;
+			if ((findChildrenByParentId && precedingClosingBraceCount > 1) ||
```

Comment:
You can simplify this condition to `!findChildrenByParentId || precedingClosingBraceCount >1`